addons/website_quote/controllers/main.py

```diff
@@ -18,50 +15,53 @@ class sale_quote(http.Controller):
+            order = SaleOrder.browse(order_id)
+        now = fields.Date.from_string(fields.Date.today())
+        action_id = request.env.ref('sale.action_quotations').id
         if token:
             if token != order.access_token:
                 return request.website.render('website.404')
             # Log only once a day
-            if request.session.get('view_quote',False)!=now:
+            if request.session.get('view_quote', False) != now:
                 request.session['view_quote'] = now
-                body=_('Quotation viewed by customer')
+                body = _('Quotation viewed by customer')
                 self.__message_post(body, order_id, message_type='comment')
         days = 0
         if order.validity_date:
-            days = (datetime.datetime.strptime(order.validity_date, '%Y-%m-%d') - datetime.datetime.now()).days + 1
+            days = (fields.Datetime.from_string(order.validity_date) - fields.Datetime.from_string(fields.Datetime.now())).days + 1
         if pdf:
-            report_obj = request.registry['report']
-            pdf = report_obj.get_pdf(request.cr, SUPERUSER_ID, [order_id], 'website_quote.report_quote', data=None, context=request.context)
+            pdf = request.env['report'].sudo().get_pdf(order, 'website_quote.report_quote')
             pdfhttpheaders = [('Content-Type', 'application/pdf'), ('Content-Length', len(pdf))]
             return request.make_response(pdf, headers=pdfhttpheaders)
-        user = request.registry['res.users'].browse(request.cr, SUPERUSER_ID, request.uid, context=request.context)
-        tx_id = request.registry['payment.transaction'].search(request.cr, SUPERUSER_ID, [('reference', '=', order.name)], context=request.context)
-        tx = request.registry['payment.transaction'].browse(request.cr, SUPERUSER_ID, tx_id, context=request.context) if tx_id else False
+        user_sudo = request.env.user.sudo()
+        payment_transaction = request.env['payment.transaction'].sudo().search([('reference', '=', order.name)])
         values = {
             'quotation': order,
             'message': message and int(message) or False,
-            'option': bool(filter(lambda x: not x.line_id, order.options)),
-            'order_valid': (not order.validity_date) or (now <= order.validity_date),
-            'days_valid': days,
-            'action': action,
-            'breadcrumb': user.partner_id == order.partner_id,
-            'tx_id': tx_id,
-            'tx_state': tx.state if tx else False,
-            'tx_post_msg': tx.acquirer_id.post_msg if tx else False,
-            'need_payment': not tx_id and order.state == 'manual'
+            'option': bool(order.options.filtered(lambda x: not x.line_id)),
+            'order_valid': (not order.validity_date) or (now <= fields.Date.from_string(order.validity_date)),
+            'days_valid': max(days, 0),
+            'action': action_id,
+            'breadcrumb': user_sudo.partner_id == order.partner_id,
```

Comment:
where it is used in <template id="so_quotation" name="Product Quotation"> ? i don't found breadcrumb in template ? if not used can we remove this and related code ?